# DemandRegio Disaggregator

The DamendRegio project aims to disaggregate the final energy consumption of the sectors:
- Industry
- Commercial, Trade and Services (CTS)
- Private Household
into a high spacial and temporal resolution.

This project includes:
- The energy carriers: power, gas, petrol products for the industry and CTS sector
- Electricity consumption of the private households for electric vehicles


## Structure of the Disaggregator
### General
The Disaggregator is structured in the following way:
- src/pipeline/*: Contains the main functions of the Disaggregator. Pipeline functions combining the data_processing functions to generate wanted the data
- src/data_processing/*: Contains the data manipulation functions
- src/configs/*: Contains the configuration files and mappings
- data/raw/*: Contains the raw input data
- src/data_access/*: Contains the data access functions (API client and local file reader)
- src/utils/*: Contains the utility and execution functions
- data/processed/*: Contains cached data to avoid recomputing the data
- data/output/*: Contains the output data

# Pipelines
src/pipeline/*: Contains the main functions of the Disaggregator. Pipeline functions combining the data_processing functions to generate wanted the data

## Consumption: `src/pipeline/pipe_consumption.py`:
This files contain the functionalitie to disaggregate the consumption on a level of industry sectors and regional_ids.
- `get_consumption_data()`: Get the consumption data for a specific year and specific energy carrier.
- `get_consumption_data_per_indsutry_sector_energy_carrier()`: Get the consumption data for a specific year and specific energy carrier for a specific industry sector (CTS or industry).

In [1]:
from src.pipeline.pipe_consumption import (
    get_consumption_data,
    get_consumption_data_per_indsutry_sector_energy_carrier,
)

In [2]:
df_consumption = get_consumption_data(
    year=2020,  # 2000-2050
    energy_carrier="power",
)
df_consumption.head()
# df_consumption["6633"]

2025-10-23 11:17:58 [INFO] pipe_consumption.py(l.33): get_consumption_data() - Loading from cache: get_consumption_data(year=2020, energy_carrier=power)


1001        1002         1003         1004  \
industry_sector                                                     
1                253.149329  982.522461  1709.037750  3486.629778   
2                  0.000000   90.717787    81.894984     0.000000   
3                  0.000000    0.000000   315.668319     0.000000   
5                  0.000000    0.000000     0.000000     0.000000   
6                  0.000000    0.000000     0.000000     0.000000   

                         1051          1053          1054          1055  \
industry_sector                                                           
1                29398.331952  13472.036762  21799.861720  16781.412960   
2                   34.760299   1821.156066     21.694408    154.095068   
3                 1015.144623     41.276045   1654.499206    436.316712   
5                    0.000000      0.000000      0.000000      0.000000   
6                54387.454756      0.000000      0.000000      0.000000   

                         1056          1057  ...         16068         16069  \
industry_sector                              ...                               
1                39131.020052  12288.629609  ...  19699.501015  11750.283827   
2                  156.556373    961.980539  ...     57.788012    563.595196   
3                  127.712240    285.841887  ...      0.000000     33.003891   
5                    0.000000      0.000000  ...      0.000000      0.000000   
6                    0.000000      0.000000  ...      0.000000      0.000000   

                        16070         16071        16072         16073  \
industry_sector                                                          
1                 8633.322821  15784.516141  3070.453742  12254.463894   
2                  654.689946    286.044316  2211.525599    466.481675   
3                   39.560170      0.000000     0.000000     65.128850   
5                32224.747179      0.000000     0.000000      0.000000   
6                    0.000000      0.000000     0.000000      0.000000   

                        16074         16075         16076         16077  
industry_sector                                                          
1                23164.220949  23203.456496  15958.961719  12566.322636  
2                   58.813836   1427.029369    310.291250      0.000000  
3                   50.633870    124.098836      0.000000      0.000000  
5                    0.000000      0.000000      0.000000      0.000000  
6                    0.000000      0.000000      0.000000      0.000000  

[5 rows x 400 columns]

In [3]:
df_filtered = get_consumption_data_per_indsutry_sector_energy_carrier(
    year=2020,
    cts_or_industry="industry",
    energy_carrier="power",
)
df_filtered.head()

2025-10-23 11:17:58 [INFO] consumption.py(l.35): get_ugr_data_ranges() - src.data_access.local_reader.get_ugr_data: Getting UGR data for year 2020
2025-10-23 11:17:58 [INFO] consumption.py(l.453): get_regional_energy_consumption() - Regional energy consumption of 2017 was used for calibration of industrial energy consumption.
2025-10-23 11:17:58 [INFO] api_reader.py(l.33): get_manufacturing_energy_consumption() - Fetching manufacturing energy consumption for year 2017
2025-10-23 11:17:58 [INFO] openffe_client.py(l.66): read_from_cache() - Reading from cache: data/api_cache/open_ffe/demandregio_demandregio_spatial_id_spatial_15_year_2017.json
2025-10-23 11:17:59 [INFO] pipe_consumption.py(l.58): get_consumption_data() - Saving consumption data power for year 2020 to cache...
2025-10-23 11:17:59 [INFO] pipe_consumption.py(l.68): get_consumption_data() - Cached: get_consumption_data(year=2020, energy_carrier=power saved to data/output/consumption/consumption_data/con_2020_power.csv


regional_id      1001        1002        1003        1004          1051   \
industry_sector                                                            
5                  0.0    0.000000    0.000000    0.000000      0.000000   
6                  0.0    0.000000    0.000000    0.000000  54387.454756   
7                  0.0    0.000000    0.000000    0.000000      0.000000   
8                  0.0  489.115703  394.686305  981.750751   4284.006346   
9                  0.0    0.000000    0.000000    0.000000      0.000000   

regional_id          1053        1054         1055         1056        1057   \
industry_sector                                                                
5                  0.00000    0.000000     0.000000     0.000000    0.000000   
6                  0.00000    0.000000     0.000000     0.000000    0.000000   
7                  0.00000    0.000000     0.000000     0.000000    0.000000   
8                298.83488  361.011472  1543.454755  1010.805686  344.650039   
9                  0.00000    0.000000     0.000000     0.000000    0.000000   

regional_id      ...        16068       16069         16070       16071  \
industry_sector  ...                                                      
5                ...     0.000000    0.000000  32224.747179    0.000000   
6                ...     0.000000    0.000000      0.000000    0.000000   
7                ...     0.000000    0.000000      0.000000    0.000000   
8                ...  1017.147024  683.808803   1015.804808  558.520581   
9                ...     0.000000    0.000000      0.000000    0.000000   

regional_id            16072        16073       16074        16075  \
industry_sector                                                      
5                   0.000000     0.000000    0.000000     0.000000   
6                   0.000000     0.000000    0.000000     0.000000   
7                   0.000000     0.000000    0.000000     0.000000   
8                2152.278454  3504.580732  224.575821  2196.020062   
9                   0.000000  4557.282614    0.000000     0.000000   

regional_id           16076        16077  
industry_sector                           
5                  0.000000     0.000000  
6                  0.000000     0.000000  
7                  0.000000     0.000000  
8                413.294576  2781.500654  
9                537.392284  1342.520660  

[5 rows x 400 columns]

In [4]:
df_kassel = df_filtered[6633]
df_kassel.head(20)

industry_sector
5          0.000000
6          0.000000
7          0.000000
8       3507.264286
9          0.000000
10     57239.022710
11     10181.218234
12         0.000000
13     23349.027839
14     34236.262052
15         0.000000
16     14992.080754
17         0.000000
18      2265.396485
19         0.000000
20    100373.112630
21         0.000000
22     54646.206374
23     47718.236360
24    111503.607461
Name: 6633, dtype: float64

## Temporal: `src/pipeline/pipe_temporal.py`:
Contains the functionalities to disaggregate the consumption on a level of temporal resolution.
`disaggregate_temporal(...)`: Disaggregates the results from the application pipeline to a temporal resolution. Differentiating between the different sectors and energy carriers.

In [15]:
%%time

from src.pipeline.pipe_temporal import disaggregate_temporal

df_temporal = disaggregate_temporal(
    sector="industry",
    energy_carrier="power",
    year=2020,
)
df_temporal.head()

2025-10-23 15:07:34 [INFO] pipe_temporal.py(l.43): disaggregate_temporal() - Loading from cache: disaggregate_temporal(sector=industry, energy_carrier=power, year=2020)


CPU times: user 5.47 s, sys: 452 ms, total: 5.92 s
Wall time: 6.07 s


regional_id         1001                                               \
industry_sector        5    6    7    8    9        10        11   12   
Date                                                                    
2020-01-01 00:00:00  0.0  0.0  0.0  0.0  0.0  0.027236  0.031103  0.0   
2020-01-01 00:15:00  0.0  0.0  0.0  0.0  0.0  0.027236  0.031103  0.0   
2020-01-01 00:30:00  0.0  0.0  0.0  0.0  0.0  0.027236  0.031103  0.0   
2020-01-01 00:45:00  0.0  0.0  0.0  0.0  0.0  0.027236  0.031103  0.0   
2020-01-01 01:00:00  0.0  0.0  0.0  0.0  0.0  0.027236  0.031103  0.0   

regional_id                              ...     16077                      \
industry_sector            13        14  ...        24        25        26   
Date                                     ...                                 
2020-01-01 00:00:00  0.005465  0.018962  ...  0.521085  0.203995  0.044498   
2020-01-01 00:15:00  0.005465  0.018962  ...  0.521085  0.203995  0.044498   
2020-01-01 00:30:00  0.005465  0.018962  ...  0.521085  0.203995  0.044498   
2020-01-01 00:45:00  0.005465  0.018962  ...  0.521085  0.203995  0.044498   
2020-01-01 01:00:00  0.005465  0.018962  ...  0.521085  0.203995  0.044498   

regional_id                                                           \
industry_sector           27        28        29        30        31   
Date                                                                   
2020-01-01 00:00:00  0.01555  0.215213  0.289197  0.057295  0.008709   
2020-01-01 00:15:00  0.01555  0.215213  0.289197  0.057295  0.008709   
2020-01-01 00:30:00  0.01555  0.215213  0.289197  0.057295  0.008709   
2020-01-01 00:45:00  0.01555  0.215213  0.289197  0.057295  0.008709   
2020-01-01 01:00:00  0.01555  0.215213  0.289197  0.057295  0.008709   

regional_id                              
industry_sector            32        33  
Date                                     
2020-01-01 00:00:00  0.060291  0.023808  
2020-01-01 00:15:00  0.060291  0.023808  
2020-01-01 00:30:00  0.060291  0.023808  
2020-01-01 00:45:00  0.060291  0.023808  
2020-01-01 01:00:00  0.060291  0.023808  

[5 rows x 11600 columns]

In [18]:
df_kassel = df_temporal["6633"]
df_kassel.head()

industry_sector        5    6    7         8    9        10        11   12  \
Date                                                                         
2020-01-01 00:00:00  0.0  0.0  0.0  0.053492  0.0  1.074217  0.168348  0.0   
2020-01-01 00:15:00  0.0  0.0  0.0  0.053492  0.0  1.074217  0.168348  0.0   
2020-01-01 00:30:00  0.0  0.0  0.0  0.053492  0.0  1.074217  0.168348  0.0   
2020-01-01 00:45:00  0.0  0.0  0.0  0.053492  0.0  1.074217  0.168348  0.0   
2020-01-01 01:00:00  0.0  0.0  0.0  0.053492  0.0  1.074217  0.168348  0.0   

industry_sector            13        14  ...        24        25        26  \
Date                                     ...                                 
2020-01-01 00:00:00  0.438196  0.642519  ...  3.165552  0.797312  0.090277   
2020-01-01 00:15:00  0.438196  0.642519  ...  3.165552  0.797312  0.090277   
2020-01-01 00:30:00  0.438196  0.642519  ...  3.165552  0.797312  0.090277   
2020-01-01 00:45:00  0.438196  0.642519  ...  3.165552  0.797312  0.090277   
2020-01-01 01:00:00  0.438196  0.642519  ...  3.165552  0.797312  0.090277   

industry_sector            27        28        29        30        31  \
Date                                                                    
2020-01-01 00:00:00  0.480312  0.269694  5.818254  0.003027  0.012106   
2020-01-01 00:15:00  0.480312  0.269694  5.818254  0.003027  0.012106   
2020-01-01 00:30:00  0.480312  0.269694  5.818254  0.003027  0.012106   
2020-01-01 00:45:00  0.480312  0.269694  5.818254  0.003027  0.012106   
2020-01-01 01:00:00  0.480312  0.269694  5.818254  0.003027  0.012106   

industry_sector            32        33  
Date                                     
2020-01-01 00:00:00  0.076958  0.380997  
2020-01-01 00:15:00  0.076958  0.380997  
2020-01-01 00:30:00  0.076958  0.380997  
2020-01-01 00:45:00  0.076958  0.380997  
2020-01-01 01:00:00  0.076958  0.380997  

[5 rows x 29 columns]

## Heat/Fuel Switch Pipeline: `src/pipeline/pipe_heat.py`:
Handles the transition from fossil fuels (gas and petrol) to clean energy carriers (electricity and hydrogen) for CTS and industry sectors.

Main function:
- `temporal_elec_load_from_fuel_switch()`: Main entry point that calculates temporal electricity demand profiles needed to replace gas/petrol consumption

In [6]:
from src.pipeline.pipe_heat import (
    temporal_elec_load_from_fuel_switch,
)

# Calculates the electricity demand that is needed to replace the gas or petrol consumption.
df_elec_fuel_switch = temporal_elec_load_from_fuel_switch(
    year=2030,
    state="HE",
    energy_carrier="petrol",
    sector="industry",
    switch_to="power",  # power or hydrogen
)

2025-10-23 11:18:18 [INFO] pipe_heat.py(l.84): temporal_elec_load_from_fuel_switch() - Load temporal_elec_load_from_fuel_switch from cache for year: 2030, state: HE, sector: industry, energy_carrier: petrol, switch_to: power


In [7]:
df_elec_fuel_switch.head()

regional_id          6411             6411.1  \
0      industry_sector            10                 10   
1          application     hot_water  mechanical_energy   
2  2030-01-01 00:00:00  0.0009805826       0.0018202821   
3  2030-01-01 00:15:00  0.0009805826       0.0018202821   
4  2030-01-01 00:30:00  0.0009805826       0.0018202821   

                     6411.2                    6411.3  \
0                        10                        10   
1  process_heat_100_to_200C  process_heat_200_to_500C   
2              0.0710481785              0.0182561484   
3              0.0710481785              0.0182561484   
4              0.0710481785              0.0182561484   

                    6411.4         6411.5        6411.6             6411.7  \
0                       10             10            11                 11   
1  process_heat_below_100C  space_heating     hot_water  mechanical_energy   
2             0.0572351113   0.0166278924  0.0000492302       0.0000913873   
3             0.0572351113   0.0166278924  0.0000492302       0.0000913873   
4             0.0572351113   0.0166278924  0.0000492302       0.0000913873   

                     6411.8  ...           6636.126                  6636.127  \
0                        11  ...                 33                        33   
1  process_heat_100_to_200C  ...  mechanical_energy  process_heat_100_to_200C   
2              0.0035669750  ...       0.0000494275              0.0010230566   
3              0.0035669750  ...       0.0000494275              0.0010230566   
4              0.0035669750  ...       0.0000494275              0.0010230566   

                   6636.128                 6636.129       6636.130  \
0                        33                       33             33   
1  process_heat_200_to_500C  process_heat_below_100C  space_heating   
2              0.0003956818             0.0001372032   0.0015044400   
3              0.0003956818             0.0001372032   0.0015044400   
4              0.0003956818             0.0001372032   0.0015044400   

            6636.131                  6636.132                  6636.133  \
0                  8                         8                         8   
1  mechanical_energy  process_heat_100_to_200C  process_heat_200_to_500C   
2       0.0000110932              0.0005551478              0.0002052368   
3       0.0000110932              0.0005551478              0.0002052368   
4       0.0000110932              0.0005551478              0.0002052368   

                  6636.134       6636.135  
0                        8              8  
1  process_heat_below_100C  space_heating  
2             0.0000711661   0.0001415011  
3             0.0000711661   0.0001415011  
4             0.0000711661   0.0001415011  

[5 rows x 3379 columns]

In other words, 0.000980 MW of electricity is needed to replace the petrol consumption that used for heating water in industry sector "10" in region 6411 at midnight on January 1st, 2030.

In [8]:
city = "6633"  # Kassel
cols_city = [col for col in df_elec_fuel_switch.columns if str(col).startswith(city)]
df_city = df_elec_fuel_switch[cols_city]
df_city.head()

6633             6633.1                    6633.2  \
0            10                 10                        10   
1     hot_water  mechanical_energy  process_heat_100_to_200C   
2  0.0007277531       0.0013284096              0.0521419221   
3  0.0007277531       0.0013284096              0.0521419221   
4  0.0007277531       0.0013284096              0.0521419221   

                     6633.3                   6633.4         6633.5  \
0                        10                       10             10   
1  process_heat_200_to_500C  process_heat_below_100C  space_heating   
2              0.0133230139             0.0421978577   0.0302220714   
3              0.0133230139             0.0421978577   0.0302220714   
4              0.0133230139             0.0421978577   0.0302220714   

         6633.6             6633.7                    6633.8  \
0            11                 11                        11   
1     hot_water  mechanical_energy  process_heat_100_to_200C   
2  0.0001053549       0.0001923104              0.0075484513   
3  0.0001053549       0.0001923104              0.0075484513   
4  0.0001053549       0.0001923104              0.0075484513   

                     6633.9  ...           6633.108                  6633.109  \
0                        11  ...                 33                        33   
1  process_heat_200_to_500C  ...  mechanical_energy  process_heat_100_to_200C   
2              0.0019287383  ...       0.0003670316              0.0075803372   
3              0.0019287383  ...       0.0003670316              0.0075803372   
4              0.0019287383  ...       0.0003670316              0.0075803372   

                   6633.110                 6633.111       6633.112  \
0                        33                       33             33   
1  process_heat_200_to_500C  process_heat_below_100C  space_heating   
2              0.0029382002             0.0010163670   0.0110343934   
3              0.0029382002             0.0010163670   0.0110343934   
4              0.0029382002             0.0010163670   0.0110343934   

            6633.113                  6633.114                  6633.115  \
0                  8                         8                         8   
1  mechanical_energy  process_heat_100_to_200C  process_heat_200_to_500C   
2       0.0000354656              0.0017725328              0.0006561545   
3       0.0000354656              0.0017725328              0.0006561545   
4       0.0000354656              0.0017725328              0.0006561545   

                  6633.116       6633.117  
0                        8              8  
1  process_heat_below_100C  space_heating  
2             0.0002269736   0.0004468369  
3             0.0002269736   0.0004468369  
4             0.0002269736   0.0004468369  

[5 rows x 118 columns]

In [9]:
# industries in city
industries = df_city.iloc[0, :].unique()
industries

array(['10', '11', '13', '14', '16', '18', '20', '22', '23', '24', '25',
       '26', '27', '28', '29', '30', '31', '32', '33', '8'], dtype=object)

In [10]:
# sector 10
# Get column names for sector 10
sector = "10"
sector_10_cols = df_city.columns[df_city.iloc[0] == sector]
sector_10 = df_city[sector_10_cols]
sector_10.head()

6633             6633.1                    6633.2  \
0            10                 10                        10   
1     hot_water  mechanical_energy  process_heat_100_to_200C   
2  0.0007277531       0.0013284096              0.0521419221   
3  0.0007277531       0.0013284096              0.0521419221   
4  0.0007277531       0.0013284096              0.0521419221   

                     6633.3                   6633.4         6633.5  
0                        10                       10             10  
1  process_heat_200_to_500C  process_heat_below_100C  space_heating  
2              0.0133230139             0.0421978577   0.0302220714  
3              0.0133230139             0.0421978577   0.0302220714  
4              0.0133230139             0.0421978577   0.0302220714

## Applications: `src/pipeline/pipe_applications.py`:
Contains the functionalities to disaggregate the consumption on a level of applications.

`disagg_applications_efficiency_factor()`: Dissaggregate the consumption data based on the applications and apply efficiency enhancement factors. The function for the effect is in `src/data_processing/effects.py` and called `apply_efficiency_factor()`.

In [11]:
from src.pipeline.pipe_applications import disagg_applications_efficiency_factor

df_applications = disagg_applications_efficiency_factor(
    sector="industry",
    energy_carrier="power",
    year=2020,
)
df_applications.head()

2025-10-23 11:18:30 [INFO] pipe_applications.py(l.43): disagg_applications_efficiency_factor() - Loading from cache: disagg_applications_efficiency_factor(sector=industry, energy_carrier=power, year=2020)


industry_sector        5                                                     \
application     lighting information_communication_technology space_cooling   
regional_id                                                                   
1001                 0.0                                  0.0           0.0   
1002                 0.0                                  0.0           0.0   
1003                 0.0                                  0.0           0.0   
1004                 0.0                                  0.0           0.0   
1051                 0.0                                  0.0           0.0   

industry_sector                                                            \
application     process_cooling mechanical_energy space_heating hot_water   
regional_id                                                                 
1001                        0.0               0.0           0.0       0.0   
1002                        0.0               0.0           0.0       0.0   
1003                        0.0               0.0           0.0       0.0   
1004                        0.0               0.0           0.0       0.0   
1051                        0.0               0.0           0.0       0.0   

industry_sector                                                   \
application     process_heat_below_100C process_heat_100_to_200C   
regional_id                                                        
1001                                0.0                      0.0   
1002                                0.0                      0.0   
1003                                0.0                      0.0   
1004                                0.0                      0.0   
1051                                0.0                      0.0   

industry_sector                           ...  \
application     process_heat_200_to_500C  ...   
regional_id                               ...   
1001                                 0.0  ...   
1002                                 0.0  ...   
1003                                 0.0  ...   
1004                                 0.0  ...   
1051                                 0.0  ...   

industry_sector                                   33                \
application     information_communication_technology space_cooling   
regional_id                                                          
1001                                       70.107726     16.453854   
1002                                      553.721958    129.955153   
1003                                      454.871891    106.755648   
1004                                      207.575163     48.716620   
1051                                     1672.434115    392.510047   

industry_sector                                                             \
application     process_cooling mechanical_energy space_heating  hot_water   
regional_id                                                                  
1001                   5.007695        356.261708      4.292310   2.861540   
1002                  39.551568       2813.811584     33.901344  22.600896   
1003                  32.490849       2311.491856     27.849299  18.566200   
1004                  14.826797       1054.820727     12.708683   8.472456   
1051                 119.459580       8498.695808    102.393925  68.262617   

industry_sector                                                   \
application     process_heat_below_100C process_heat_100_to_200C   
regional_id                                                        
1001                           5.640960                36.246171   
1002                          44.553202               286.278019   
1003                          36.599594               235.171862   
1004                          16.701772               107.317771   
1051                         134.566263               864.659814   

industry_sector                                                   
application   

In [12]:
df_region = df_applications.loc[9162]  # München
df_region.loc["29"].head(12)  # cars, trucks, vans

application
lighting                                 45088.920760
information_communication_technology     38760.651180
space_cooling                            13447.572858
process_cooling                              0.000000
mechanical_energy                       373367.905240
space_heating                             3322.341530
hot_water                                 2214.894353
process_heat_below_100C                   9348.580062
process_heat_100_to_200C                 31161.933540
process_heat_200_to_500C                 10906.676739
process_heat_above_500C                      0.000000
Name: 9162, dtype: float64

We can see that 45088.920760 MWh is the amount of electrical energy consumed specifically for lighting purposes in industry sector 29 (majufacturing of motor vehicles) in region 9162 (München) for 2020, after accounting for efficiency improvements.

Here's how this value is derived:

1. Base consumption data: The function starts with total power consumption for industry sector 29 in region 9162
2. Application disaggregation: Using get_application_dissaggregation_factors(), it applies decomposition factors that split the total consumption into different applications (lighting, mechanical energy, process heat, etc.)
3. Efficiency factors: The apply_efficiency_factor() function then adjusts these values based on year-specific efficiency improvements

## EV regional consumption: `src/pipeline/pipe_ev_regional_consumption.py`:
Contains two approaches to disaggregate and project the power consumption of electric vehicles in private households on a level of 400 regional_ids.
The KBA approaches (also referred as s1 and s2) are calculating the consumption based on: vehicle stock data * average km driven by EVs * average mwh per km. For the historical calculations both KBA approaches using the same data.The KBA_1 (s1) approach uses the normative goal (15mio EVs by 2030 and only EVs by 2045) and the KBA_2 (s2) approach uses the projected number of EVs from literature to estimate the future vehicle stock.

In [13]:
from src.pipeline.pipe_ev_regional_consumption import (
    electric_vehicle_consumption_by_regional_id,
)

df_ev_consumption = electric_vehicle_consumption_by_regional_id(
    year=2020,
    szenario="KBA_1",  # KBA_1, KBA_2, UGR
)
df_ev_consumption.head()

2025-10-23 11:18:30 [INFO] pipe_ev_regional_consumption.py(l.358): electric_vehicle_consumption_by_regional_id() - Load electric_vehicle_consumption_by_regional_id from cache for year: 2020, szenario: KBA_1, s2_szenario: None


power[mwh]
regional_id              
8111         11621.672710
8115          9692.555250
8116          7897.029162
8117          3030.491961
8118          8032.675097

## EV temporal consumption: `src/pipeline/pipe_ev_temporal.py`:
Contains the functionalities to disaggregate the consumption on a level of temporal resolution.
KBA approaches are using the charging profiles of all locations (home, work, public), since the EVs can be charged at all locations.
The UGR approach is using the charging profiles of the home location since it already only contains the electricity demand in private households.

In [14]:
from src.pipeline.pipe_ev_temporal import (
    electric_vehicle_consumption_by_region_id_and_temporal_resolution,
)

df_ev_temporal = electric_vehicle_consumption_by_region_id_and_temporal_resolution(
    year=2020,
    szenario="KBA_1",  # KBA_1, KBA_2, UGR
)
df_ev_temporal.head()

2025-10-23 11:18:30 [INFO] pipe_ev_temporal.py(l.67): electric_vehicle_consumption_by_region_id_and_temporal_resolution() - Load electric_vehicle_consumption_by_regional_id from cache for year: 2020, szenario: KBA_1, s2_szenario: None


1001                  1001.1  \
regional_id                                                          
charging_location            home_charging           work_charging   
datetime                               NaN                     NaN   
2020-01-01 00:00:00   0.008113626138405702  0.00018346739136863554   
2020-01-01 00:10:00  0.0077798423944483224  0.00015835749587339717   
2020-01-01 00:20:00   0.007484220511729203  0.00012396222186568409   

                                    1001.2                  1002  \
regional_id                                                        
charging_location          public_charging         home_charging   
datetime                               NaN                   NaN   
2020-01-01 00:00:00  6.453806671950755e-05   0.01957739876987581   
2020-01-01 00:10:00  5.252123928886702e-05  0.018772010729203804   
2020-01-01 00:20:00  3.093444701037098e-05  0.018058703585841773   

                                     1002.1                  1002.2  \
regional_id                                                           
charging_location             work_charging         public_charging   
datetime                                NaN                     NaN   
2020-01-01 00:00:00  0.00044268915289193096  0.00015572414188816947   
2020-01-01 00:10:00   0.0003821013924018001  0.00012672869416291118   
2020-01-01 00:20:00  0.00029910890746824386   7.464184256419615e-05   

                                     1003                 1003.1  \
regional_id                                                        
charging_location           home_charging          work_charging   
datetime                              NaN                    NaN   
2020-01-01 00:00:00   0.01566151532011465  0.0003541421938410837   
2020-01-01 00:10:00  0.015017221495082562  0.0003056732347990015   
2020-01-01 00:20:00  0.014446590489144153   0.000239280958198839   

                                     1003.2                  1004  ...  \
regional_id                                                        ...   
charging_location           public_charging         home_charging  ...   
datetime                                NaN                   NaN  ...   
2020-01-01 00:00:00  0.00012457610240059204  0.010361780231913186  ...   
2020-01-01 00:10:00  0.00010138034212106658  0.009935510430855882  ...   
2020-01-01 00:20:00  5.9711934899110166e-05   0.00955797652329999  ...   

                                    16074.2                  16075  \
regional_id                                                          
charging_location           public_charging          home_charging   
datetime                                NaN                    NaN   
2020-01-01 00:00:00   3.142970981365331e-05  0.0032921765462669613   
2020-01-01 00:10:00  2.5577576054096174e-05  0.0031567407997047485   
2020-01-01 00:20:00  1.5064918151443717e-05  0.0030367895704702195   

                                   16075.1                 16075.2  \
regional_id                                                          
charging_location            work_charging         public_charging   
datetime                               NaN                     NaN   
2020-01-01 00:00:00  7.444353887709307e-05   2.618689917072329e-05   
2020-01-01 00:10:00  6.425497366365057e-05  2.1310963706994134e-05   
2020-01-01 00:20:00  5.029878287312088e-05  1.2551929209212119e-05   

                                     16076                 16076.1  \
regional_id                                                          
charging_location            home_charging           work_charging   
datetime                               NaN                     NaN   
2020-01-01 00:00:00   0.003746125294183599   8.470834417802957e-05   
2020-01-01 00:10:00  0.0035920147023598865   7.311490703359853e-05   
2020-01-01 00:20:00  0.0034555237433882575  5.7234337265814283e-05   

                                    16076.2                  16077  \
regional_id                      